src/commands/sourcemaps/git.ts

```diff
@@ -0,0 +1,242 @@
+import fs from 'fs'
+import * as simpleGit from 'simple-git'
+import {Writable} from 'stream'
+import {URL} from 'url'
+import {promisify} from 'util'
+
+import {renderGitError, renderProjectPathNotFoundWarning, renderSourcesNotFoundWarning} from './renderer'
+
+// Returns a configured SimpleGit.
+export const newSimpleGit = (): simpleGit.SimpleGit => {
+  const options: simpleGit.SimpleGitOptions = {
+    baseDir: process.cwd(),
+    binary: 'git',
+    // We are invoking at most 3 git commands at the same time.
+    maxConcurrentProcesses: 3,
+  }
+
+  return simpleGit.gitP(options)
+}
+
+// Returns the remote of the current repository.
+export const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
+  const remotes = await git.getRemotes(true)
+  if (remotes.length === 0) {
+    throw new Error('No git remotes available')
+  }
+
+  for (const remote of remotes) {
+    // We're trying to pick the remote called with the default git name 'origin'.
+    if (remote.name === 'origin') {
+      return remote.refs.push
+    }
+  }
+
+  // Falling back to picking the first remote in the list if 'origin' is not found.
+  return remotes[0].refs.push
+}
+
+// StripCredentials removes credentials from a remote HTTP url.
+export const stripCredentials = (remote: string) => {
+  try {
+    const url = new URL(remote)
+    url.username = ''
+    url.password = ''
+
+    return url.toString()
+  } catch {
+    return remote
+  }
+}
+
+// Returns the hash of the current repository.
+const gitHash = async (git: simpleGit.SimpleGit): Promise<string> => git.revparse('HEAD')
+
+// Returns the tracked files of the current repository.
+export const gitTrackedFiles = async (git: simpleGit.SimpleGit): Promise<string[]> => {
+  const files = await git.raw('ls-files')
```

Comment:
Is this command run from current location or from the base of the repository?

Does it work correctly if the command `datadog-ci` is run from a subfolder?

Reply:
Good point. The output if ran is a subfolder would be different. Changed it to make sure the baseDir is set to the root of the repo.

Reply:
It is now covered in e832d79 cc @lraucy